### PR TITLE
Add wdio screenshot testing

### PR DESCRIPTION
**Pull request by alisman**
# What? Why?
Fix # .

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: end-to-end-tests/specs/screenshot.spec.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+var assert = require('assert');
+var expect = require('chai').expect;
+
+const CBIOPORTAL_URL = process.env.CBIOPORTAL_URL.replace(/\/$/, "");
+
+function ssAssert(result, message){
+    assert(result[0].isWithinMisMatchTolerance);
+};
+
+describe('fdas', function(){
```
Comment (adamabeshouse) [inline end-to-end-tests/specs/screenshot.spec.js]: fdas?

File: end-to-end-tests/wdio/webdriver-manager.conf.js [hunk 2]
```diff
@@ -1,3 +1,21 @@
+var path = require('path');
+var VisualRegressionCompare = require('wdio-visual-regression-service/compare');
+var getScreenshotName = require('./getScreenshotName');
+
+// function getScreenshotName(basePath) {
```
Comment (adamabeshouse) [inline end-to-end-tests/wdio/webdriver-manager.conf.js]: remove commented code?

File: end-to-end-tests/specs/screenshot.spec.js (hunk 1 above)
Comment (alisman) [reply end-to-end-tests/specs/screenshot.spec.js]: yeah, i just needed to run these on travis to see how it handled the screenshots.  will fix this up.

**Comment by inodb** [comment]
@alisman i think this fails on travis, because current cbioportal.org backend doesn't work with this frontend. On circleci it is using cbioportal.org/beta, so that's why those pass (i should make them point to the same URL)

**Comment by inodb** [comment]
@alisman prolly good idea to remove the user/key from the config file. Instead do:

```
export BROWSERSTACK_USERNAME='xxxx' BROWSERSTACK_ACCESS_KEY='yyyy'
```
before running the tests and keep original config

**Comment by alisman** [comment]
@inodb thanks.  yes, that makes sense.

**Comment by alisman** [comment]
gotta fix that one screensehot

File: end-to-end-tests/wdio/getScreenshotName.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    return function(context) {
+        var type = context.type;
+        var testName = context.test.title;
+        var browserVersion = parseInt(context.browser.version, 10);
+        var browserName = context.browser.name;
+        var browserViewport = context.meta.viewport;
+        var browserWidth = browserViewport.width;
+        var browserHeight = browserViewport.height;
+
+        return path.join(basePath, `${testName}_${type}_${browserName}_v${browserVersion}_${browserWidth}x${browserHeight}.png`);
```
Comment (inodb) [inline end-to-end-tests/wdio/getScreenshotName.js]: would be good to lower case this whole string and replace spaces by underscores

**Pull request closed without merging**